scripts/subscribeEventProof.js

```diff
@@ -0,0 +1,94 @@
+
+const { Api } = require('@cennznet/api');
+
+async function main (networkName, bridgeContractAddress) {
+    networkName = networkName || 'local';
+    // the API has connected to the node and completed the initialisation process
+    const rpc = {
+        ethy: {
+            subscribeEventProofs: {
+                description: 'Subscribe event proof',
+                params: [],
+                type: 'EventProof'
+            },
+            getEventProof: {
+                description: 'Get event proof for event Id',
+                params: [{
+                    "name": "EventId",
+                    "type": "EthyEventId"
+                }],
+                type: 'Option<VersionedEventProof>'
+            }
+        }}
+    const api = await Api.create({network: networkName, rpc, types: {
+        EthyEventId: 'u64',
+        EventProof: {
+            digest: '[u8; 32]',
+            eventId: 'EthyEventId',
+            validatorSetId: 'u64',
+            signatures: 'Vec<EthereumSignature>'
+        },
+            VersionedEventProof: {
+                _enum: {
+                    sentinel: null,
+                    EventProof: 'EventProof'
+                }
+            }},
+    });
+
+    const [txExecutor] = await ethers.getSigners();
+
+    // Get the bridge instance that was deployed
+    const Bridge = await ethers.getContractFactory('CENNZnetBridge');
+    console.log('Connecting to CENNZnet bridge contract...');
+    bridgeContractAddress = bridgeContractAddress || "0x75a2488b80D1a12cB0209cB1C40986863745Ee2f";
+    const bridge = await Bridge.attach(bridgeContractAddress);
+    await bridge.deployed();
+    console.log('CENNZnet bridge deployed to:', bridge.address);
+
+    await api.rpc.chain
+        .subscribeFinalizedHeads(async (head) => {
+           const blockNumber = head.number.toNumber();
+           console.log('At blocknumber:',blockNumber);
+           const blockHash = head.hash.toString();
+           const events = await api.query.system.events.at(blockHash);
+           events.map(async ({event}) => {
+                const { section, method, data } = event;
+                if (section === 'ethBridge' && method === 'AuthoritySetChange') {
+                    const eventIdFound = data.toHuman()[0];
+                    const eventProof = (await api.rpc.ethy.getEventProof(eventIdFound)).toJSON().EventProof;
```

Comment:
what happens if this comes back empty? e.g proof isn't read yet

Reply:
Once we deploy on rata.. may be I can start the script to see what happens.. I haven't really encountered a scenario where the proof isn't ready..

Reply:
it will definitely happen on real network as the proof takes sometime to be signed by all the active validators

Reply:
have updated the script.. but I am keen to test it on rata to see if it works